core/io/http_client_tcp.cpp

```diff
@@ -256,10 +259,17 @@ Error HTTPClientTCP::poll() {
 					return OK; // Still resolving
 
 				case IP::RESOLVER_STATUS_DONE: {
-					IPAddress host = IP::get_singleton()->get_resolve_item_address(resolving);
-					Error err = tcp_connection->connect_to_host(host, conn_port);
+					ip_candidates = IP::get_singleton()->get_resolve_item_addresses(resolving);
```

Comment:
`ip_candidates` should also be cleared when reaching `STATUS_CONNECTED` (i.e. fully connected, including SSL handshake).